torch/distributed/_tensor/redistribute.py

```diff
@@ -72,19 +73,25 @@ def _decompose_reshard(val: List[_PlacementItem]) -> List[_PlacementItem]:
 
 
 # Intentionally expose this API to trace ops on local tensors
-def _redistribute_with_local_tensor(
+def redistribute_local_tensor(
```

Comment:
ditto, like I mentioned in the `dispatch.py`, maybe the old name `redistribute_with_local_tensor` or redistribute_with_local_tensor_updated` is better? because  the name here seems to suggest that we directly redistribute a Tensor, which is somewhat confusing?

Reply:
hmmm this function/api is indeed directly redistribute a torch.Tensor (i.e. the first arg is the local_tensor) with src/dst dtensor spec, maybe we should add more comment on the API to clarify?

Reply:
OK, essentially there is no DTensor involved indeed. We just call collectives directly (and with no-autograd). I just think redistributing a tensor is a little confusing. Well since we redistribute a local tensor of a DTensor, this sounds make sense to me. If we can add some comments here, that would be much appreciated.

Reply:
Yep will add in the follow up PR